File: model/src/main/java/no/nav/pensjon/vtp/testmodell/inntektytelse/InntektYtelseIndeks.java
package no.nav.pensjon.vtp.testmodell.inntektytelse;

import java.util.Map;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;
import java.util.stream.Stream;

import org.springframework.stereotype.Component;

@Component
public class InntektYtelseIndeks {

    private final Map<String, InntektYtelseModell> byIdent = new ConcurrentHashMap<>();

    public Optional<InntektYtelseModell> getInntektYtelseModell(String ident) {
        return Optional.ofNullable(byIdent.get(ident));
    }

    public Stream<Map.Entry<String, InntektYtelseModell>> entries() {
        return byIdent.entrySet().stream();
    }

    public void leggTil(String ident, InntektYtelseModell iy) {
        if(ident != null && iy != null){
            byIdent.put(ident, iy);
        }
    }

    public Optional<InntektYtelseModell> getInntektYtelseModellFraAktørId(String aktørId) {
        return getInntektYtelseModell(aktørId.substring(aktørId.length() - 11));
    }
}

File: model/src/main/java/no/nav/pensjon/vtp/testmodell/inntektytelse/InntektYtelseIndeks.kt
package no.nav.pensjon.vtp.testmodell.inntektytelse

import org.springframework.stereotype.Component
import java.util.concurrent.ConcurrentHashMap

@Component
class InntektYtelseIndeks {
    private val byIdent: MutableMap<String, InntektYtelseModell> = ConcurrentHashMap()

    fun getInntektYtelseModell(ident: String): InntektYtelseModell? {
        return byIdent[ident]
    }

    fun leggTil(ident: String, iy: InntektYtelseModell) {
        byIdent[ident] = iy
    }

    fun entries(): Map<String, InntektYtelseModell> = byIdent

    fun getInntektYtelseModellFraAktørId(aktørId: String): InntektYtelseModell? {
        return getInntektYtelseModell(aktørId.substring(aktørId.length - 11))
    }
}
